lib/foreman_ansible_core/runner/playbook.rb

```diff
@@ -118,6 +118,20 @@ def add_to_known_hosts(host)
         logger.error('[foreman_ansible] - Failed to save host key for '\
           "#{host}: #{e}")
       end
+
+      def rebuild_secrets(inventory, secrets)
+        inventory['all']['hosts'].each do |name|
+          per_host = secrets['per-host'][name]
+
+          new_secrets = {
+            'ansible_ssh_pass' => inventory['ssh_password'] || per_host['ansible_ssh_pass'],
+            'ansible_sudo_pass' => inventory['sudo_password'] || per_host['ansible_sudo_pass']
+          }
+          inventory['_meta']['hostvars'][name].update(new_secrets)
+        end
+
+        inventory
+      end
```

Comment:
Is this covered in "when using secrets" test? I am not sure because it's stubbed many times.